flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/StandaloneFlinkService.java

```diff
@@ -171,7 +171,11 @@ protected void deleteClusterInternal(ObjectMeta meta, boolean deleteHaConfigmaps
 
     @Override
     public boolean scale(ObjectMeta meta, JobSpec jobSpec, Configuration conf) {
-        if (conf.get(JobManagerOptions.SCHEDULER_MODE) == null) {
+        if (conf.get(JobManagerOptions.SCHEDULER_MODE) == null
+                && conf.get(StandaloneKubernetesConfigOptionsInternal.CLUSTER_MODE)
+                        .equals(
+                                StandaloneKubernetesConfigOptionsInternal.ClusterMode
+                                        .APPLICATION)) {
```

Comment:
Instead of checking the ClusterMode, it's simpler to check `jobSpec != null`, and also please explicitly check the `SCHEDULER_MODE != REACTIVE` instead of just checking null